.github/workflows/golangci-lint.yml

```diff
@@ -13,7 +13,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: '~1.15'
+          go-version: 1.17
```

Comment:
:question: Is '1.17' equivalent to `~1.17` (which AFAIK will pick up the latest patch version on the 1.17 version)?

Reply:
It's probably OK to have any 1.17 as Go follows semver, so we wouldn't be relying on any specific patch version I think for just testing, only for releasing. That said, getting it from the gomod file is probably the most portable for the future as well, if it's not too complicated to do :+1:

Reply:
I'll probably leave it as is here and then try to improve it on main.